packages/@aws-cdk/aws-synthetics/test/canary.test.ts

```diff
@@ -0,0 +1,223 @@
+import '@aws-cdk/assert/jest';
+import { objectLike } from '@aws-cdk/assert';
+import * as iam from '@aws-cdk/aws-iam';
+import * as s3 from '@aws-cdk/aws-s3';
+import { App, Duration, Lazy, Stack } from '@aws-cdk/core';
+import * as synthetics from '../lib';
+
+let stack: Stack;
+beforeEach(() => {
+  stack = new Stack(new App(), 'canaries');
+});
+
+test('Basic canary properties work', () => {
+  // WHEN
+  new synthetics.Canary(stack, 'Canary', {
+    canaryName: 'mycanary',
+    test: synthetics.Test.custom({
+      handler: 'index.handler',
+      code: synthetics.Code.fromInline('foo'),
+    }),
+    successRetentionPeriod: Duration.days(10),
+    failureRetentionPeriod: Duration.days(10),
+    startAfterCreation: false,
+    timeToLive: Duration.minutes(30),
+    runtime: synthetics.Runtime.SYNTHETICS_1_0,
+  });
+
+  // THEN
+  expect(stack).toHaveResourceLike('AWS::Synthetics::Canary', {
+    Name: 'mycanary',
+    SuccessRetentionPeriod: 10,
+    FailureRetentionPeriod: 10,
+    StartCanaryAfterCreation: false,
+    Schedule: objectLike({ DurationInSeconds: '1800'}),
+    RuntimeVersion: 'syn-1.0',
+  });
+});
+
+test('Canary can have generated name', () => {
+  // WHEN
+  new synthetics.Canary(stack, 'Canary', {
+    test: synthetics.Test.custom({
+      handler: 'index.handler',
+      code: synthetics.Code.fromInline('foo'),
+    }),
+  });
+
+  // THEN
+  expect(stack).toHaveResourceLike('AWS::Synthetics::Canary', {
+    Name: 'canariescanary8ee009f',
+  });
+});
+
+test('Name validation does not fail when using Tokens', () => {
+  // WHEN
+  new synthetics.Canary(stack, 'Canary', {
+    canaryName: Lazy.stringValue({produce: () => 'myCanary'}),
+    test: synthetics.Test.custom({
+      handler: 'index.handler',
+      code: synthetics.Code.fromInline('foo'),
+    }),
+  });
+
+  // THEN: no exception
+  expect(stack).toHaveResourceLike('AWS::Synthetics::Canary');
+});
+
+test('Throws when name is specified incorrectly', () => {
+  expect(() => new synthetics.Canary(stack, 'Canary', {
+    canaryName: 'myCanary',
+    test: synthetics.Test.custom({
+      handler: 'index.handler',
+      code: synthetics.Code.fromInline('foo'),
+    }),
+  }))
+    .toThrowError('Canary name must be lowercase, numbers, hyphens, or underscores (no spaces) (myCanary)');
+});
+
+test('Throws when name has more than 21 characters', () => {
+  expect(() => new synthetics.Canary(stack, 'Canary', {
+    canaryName: 'a'.repeat(22),
+    test: synthetics.Test.custom({
+      handler: 'index.handler',
+      code: synthetics.Code.fromInline('foo'),
+    }),
+  }))
+    .toThrowError('Canary name is too large, must be <= 21 characters, but is 22');
```

Comment:
This error message is *slightly* misleading. It should be `between 1 and 21 characters`, not `<= 21 characters`. Right now you're telling folks an empty string would be the easiest way to go...